src/eventbridge.ts

```diff
@@ -0,0 +1,68 @@
+import { aws_events } from "aws-cdk-lib";
+import { Rule } from "aws-cdk-lib/aws-events";
+import { Construct } from "constructs";
+import { FunctionDecl } from "./declaration";
+import { synthesizeEventPattern } from "./eventpattern";
+
+export interface EventBusEvent<
+  T = any,
+  DetailType extends string = string,
+  Source extends string = string
+> {
+  source: Source;
+  detailType: DetailType;
+  detail: T;
+}
+
+type EventHandlerFunction<E extends EventBusEvent = EventBusEvent<any>> = (
+  event: E
+) => void;
+
+export type EventPredicateFunction<
+  E extends EventBusEvent = EventBusEvent<any>
+> = (event: E) => void;
+
+export class EventBusRule<T extends EventBusEvent> extends Construct {
+  public readonly rule: aws_events.Rule;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    bus: aws_events.EventBus,
+    predicate: EventPredicateFunction<T>
+  ) {
+    super(scope, id);
+
+    const decl = predicate as unknown as FunctionDecl;
+    const pattern = synthesizeEventPattern(decl);
+
+    this.rule = new Rule(this, "rule", {
+      eventPattern: pattern as aws_events.EventPattern,
+      eventBus: bus,
+    });
+  }
+}
+
+export class EventBus<E extends EventBusEvent> extends Construct {
+  /**
+   * This static property identifies this class as an EventBus to the TypeScript plugin.
+   */
+  public static readonly FunctionlessType = "EventBus";
+
+  public readonly decl: FunctionDecl;
+
+  constructor(
```

Comment:
I wonder if this class should instead just wrap the `aws_events.EventBus`? Does it need to be a Construct? That's what `Table` and `Function` do.

Reply:
It was handy to be a construct because it can be the parent of all of the rules and the eb could be from another stack/imported. I'll play around with it once I get things actually working. You mentioned the step function class is a construct?

Reply:
Yeah it is. I've been flipping and flopping. Good point on it being in a different stack, but couldn't that be achieved by making the rule a Construct?

Reply:
See latest commit. Rule is a construct, takes scope and id. Event bus is just a wrapper.

https://github.com/sam-goodwin/functionless/pull/39/commits/04bde7229e112e66382a74a8052e434b71d63ddd#diff-3bb1081149ef42a1526cc7ecaa1af0a20897ac5e897324e066afca37d8ef383dR48-R67